bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -68,39 +70,155 @@ public JdbcConfiguration(Map<Object, Object> configuration) {
+            logger.warn("JDBC::updateConfig: SQL password is missing. Attempting to connect without password. "
+                    + "To specify a password configure the jdbc:password parameter in openhab.cfg.");
+        } else if (!(url == null || StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false")) {
+            logger.debug("JDBC::updateConfig: mandatory user={}", user);
+            logger.debug("JDBC::updateConfig: mandatory password=<masked> password.length={}", password.length());
+            logger.debug("JDBC::updateConfig: mandatory url={}", url);
+
+            // set database type and database type class
+            setDBDAOClass(parsedURL.getProperty("dbShortcut")); // derby, h2, hsqldb, mariadb, mysql, postgresql,
+                                                                // sqlite
+            // set user
+            dBDAO.databaseProps.setProperty("dataSource.user", user);
+
+            // set password
+            dBDAO.databaseProps.setProperty("dataSource.password", password);
+
+            // set sql-types from external config
+            setSqlTypes();
+
+            String et = (String) configuration.get("reconnectCnt");
+            if (StringUtils.isNotBlank(et)) {
+                errReconnectThreshold = Integer.parseInt(et);
+                logger.debug("JDBC::updateConfig: errReconnectThreshold={}", errReconnectThreshold);
+            }
+
+            String np = (String) configuration.get("tableNamePrefix");
+            if (StringUtils.isNotBlank(np)) {
+                tableNamePrefix = np;
+                logger.debug("JDBC::updateConfig: tableNamePrefix={}", tableNamePrefix);
+            }
+
+            String dd = (String) configuration.get("numberDecimalcount");
+            if (StringUtils.isNotBlank(dd)) {
+                numberDecimalcount = Integer.parseInt(dd);
+                logger.debug("JDBC::updateConfig: numberDecimalcount={}", numberDecimalcount);
+            }
+
+            String rn = (String) configuration.get("tableUseRealItemNames");
+            if (StringUtils.isNotBlank(rn)) {
+                tableUseRealItemNames = "true".equals(rn) ? Boolean.parseBoolean(rn) : false;
+                logger.debug("JDBC::updateConfig: tableUseRealItemNames={}", tableUseRealItemNames);
+            }
+
+            String td = (String) configuration.get("tableIdDigitCount");
+            if (StringUtils.isNotBlank(td)) {
+                tableIdDigitCount = Integer.parseInt(td);
+                logger.debug("JDBC::updateConfig: tableIdDigitCount={}", tableIdDigitCount);
+            }
+
+            String rt = (String) configuration.get("rebuildTableNames");
+            if (StringUtils.isNotBlank(rt)) {
+                rebuildTableNames = "true".equals(rt) ? Boolean.parseBoolean(rt) : false;
+                logger.debug("JDBC::updateConfig: rebuildTableNames={}", rebuildTableNames);
+            }
+
+            // undocumented
```

Comment:
*undocumented* means, this is not documented for the *normal* user outside the code and not in wiki.
For debug/testing and helping when driver problems occur, I would like to stay this at the moment.